cmd/agent/app/server.go

```diff
@@ -133,6 +135,32 @@ func (a *Agent) runProxyConnection(o *options.GrpcProxyAgentOptions, drainCh, st
 		}),
 	}
 	cc := o.ClientSetConfig(dialOptions...)
+
+	if o.ServerLeaseSelector != "" {
+		var k8sClient *kubernetes.Clientset
+		config, err := clientcmd.BuildConfigFromFlags("", o.KubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load kubernetes client config: %v", err)
+		}
+		k8sClient, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create kubernetes clientset: %v", err)
+		}
+		serverLeaseSelector, err := labels.Parse(o.ServerLeaseSelector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid server count lease selector: %w", err)
+		}
+		serverLeaseCounter := agent.NewServerLeaseCounter(
+			k8sClient,
+			serverLeaseSelector,
+		)
+		if err != nil {
```

Comment:
This is the error from the lease selector. I think you can just remove this block?

Reply:
Oops, good catch.